src/main/resources/ColorFormats.js

```diff
@@ -1,55 +1,53 @@
+const glob = require("glob");
+const log = require("electron-log");
 class ColorFormats {
   constructor() {
-    this._colorFormats = [
-      {
-        title: "CSS Hex",
-        sub_title: "#rrggbb",
-        icon: "css",
-        value: "css_hex"
-      },
-      {
-        title: "CSS HSL",
-        sub_title: "hsl(hue, sat%, light%)",
-        icon: "css_2",
-        value: "css_hsl"
-      },
-      {
-        title: "CSS HSLA",
-        sub_title: "hsl(hue, sat%, light%, alpha)",
-        icon: "css_2",
-        value: "css_hsla"
-      },
-      {
-        title: "CSS RGB",
-        sub_title: "rgb(rrr,ggg,bbb)",
-        icon: "css_2",
-        value: "css_rgb"
-      },
-      {
-        title: "CSS RGBA",
-        sub_title: "rgb(rrr,ggg,bbb,alpha)",
-        icon: "css_2",
-        value: "css_rgba"
-      }
-    ];
+    this._colorFormats = [];
+    this._loadFormats();
     this._selectedFormat = "css_hex";
   }
 
   get formats() {
     return this._colorFormats;
   }
 
-  set formats(f) {
-    this._colorFormats = f;
-  }
-
   get selectedFormat() {
     return this._selectedFormat;
   }
 
   set selectedFormat(format) {
     this._selectedFormat = format;
   }
+
+  /**
+   * Load all format plugins from ./formats/*.js directory
+   */
+  _loadFormats() {
+    glob(__dirname + "/formats/*.js", { nodir: true }, (err, files) => {
+      if (err) {
+        log.error(err);
+      } else {
+        log.log("Plugins:", files);
+        files.forEach((pluginPath, index) => {
+          const plug = require(pluginPath);
+          const plugConfigParams = plug.config();
+          if (plugConfigParams.type === "format") {
+            this._colorFormats.push({
+              title: plugConfigParams.format.displayName,
+              sub_title: plugConfigParams.format.displayFormat,
+              icon: plugConfigParams.format.icon,
+              value: plugConfigParams.name,
+              convertFromHex: hex_color => plug.convertHexColor(hex_color)
+            });
+          } else {
+            log.error(
+              new Error("Plugin has the wrong type, expected type 'format', received '" + plugConfigParams.type + "'")
+            );
+          }
+        });
+      }
+    });
+  }
 }

```

Comment:
Maybe create a separate plugin handler to accommodate other plugins at a later date. All plugins with `type: 'format'` would be used as a format converter

Reply:
A full plugin system should be implemented at a later date